packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -106,125 +36,25 @@ function calculateContent(trace: Trace, span: Span, allSpans: Span[], resultValu
   }
   // selfTime
   let tempSelf = 0;
-  let longerAsParent = false;
-  let kinderSchneiden = false;
-  let allOverlay = [];
-  const longerAsParentSpan = [];
   if (span.hasChildren) {
-    const allChildren = [] as any;
-    for (let i = 0; i < allSpans.length; i++) {
-      // i am a child?
-      if (allSpans[i].references.length === 1) {
-        if (span.spanID === allSpans[i].references[0].spanID) {
-          allChildren.push(allSpans[i]);
-        }
-      }
-    }
-    // i only have one child
-    if (allChildren.length === 1) {
-      if (
-        span.relativeStartTime + span.duration >=
-        allChildren[0].relativeStartTime + allChildren[0].duration
-      ) {
-        tempSelf = span.duration - allChildren[0].duration;
-      } else {
-        tempSelf = allChildren[0].relativeStartTime - span.relativeStartTime;
-      }
-    } else {
-      // is the child longer as parent
-      for (let i = 0; i < allChildren.length; i++) {
-        if (
-          span.duration + span.relativeStartTime <
-          allChildren[i].duration + allChildren[i].relativeStartTime
-        ) {
-          longerAsParent = true;
-          longerAsParentSpan.push(allChildren[i]);
-        }
-      }
-      // Do the children overlap?
-      for (let i = 0; i < allChildren.length; i++) {
-        for (let j = 0; j < allChildren.length; j++) {
-          // aren't they the same kids?
-          if (allChildren[i].spanID !== allChildren[j].spanID) {
-            // if yes the children cut themselves or lie into each other
-            if (
-              allChildren[i].relativeStartTime <= allChildren[j].relativeStartTime &&
-              allChildren[i].relativeStartTime + allChildren[i].duration >= allChildren[j].relativeStartTime
-            ) {
-              kinderSchneiden = true;
-              allOverlay.push(allChildren[i]);
-              allOverlay.push(allChildren[j]);
-            }
-          }
-        }
+    const spanDRange = new DRange(span.startTime, span.startTime + span.duration);
+    const childrenDrange = new DRange();
+    span.childSpanIds.forEach(eachId => {
+      const childSpan = allSpans.find(a => a.spanID === eachId && a.references[0].refType === 'CHILD_OF');
```

Comment:
> you need to check that the reference is actually to the right parent span (it could be pointing to a different trace altogether).

Why to check again, explicitly span.childSpanIds have span ids of its children only

Reply:
yes, but span.References can point upwards to many "parent" spans (not really parent, but happened-before spans)